accepted/2021/preview-features/preview-features.md

```diff
@@ -196,6 +196,44 @@ namespace System
 </Project>
 ```
 
+### SDK support of preview features
+
+The point of preview features is to gather customer feedback and adjust course
+accordingly. This includes doing breaking changes, just like in any other
+preview.
+
+Many preview features will span tooling as well as libraries. Tooling could
+include MSBuild targets but also language and IDE features.
+
+Unlike runtime and framework, projects don't directly indicate which .NET SDK
+they are using. Instead it's implied by the CLI or by the IDE version and
+configuration (Visual Studio, Visual Studio Code, and Rider all have different
+ways how and to what extent this can be controlled by the developer).
+
+Just like in the case of runtime and framework, we don't plan to support preview
+features in the SDK across versions. For instance, a preview API in `net6.0`
+shouldn't be expected to have the same shape -- or even exists -- in `net7.0`.
+The same is true for the .NET SDK: .NET 6 SDK might have preview features in
+MSBuild and C# that don't exist or have a very different syntax in .NET 7.
+
+Since preview features between runtime, framework, and tooling are usually in
+support of each other, you can't expect to use, say, the .NET 7 SDK to build for
+.NET 6 when using preview features. This is likely counter intuitive because
```

Comment:
This sentence makes it sound like it might work when in reality it will not work. There is a 100% chance using preview features will break when you move to a new .NET SDK right?

Reply:
Well that depends. We have marked API as preview that had nothing to do with generic math/static interface members. And that's the are we're eager to expand.

Reply:
Not sure why generic math / static has anything to do with this. The break seems irrespective of whether it's a language or runtime feature. 

The core issue is that `<EnablePreviewFeatures>` only takes effect when the target framework and .NET SDK major version are the same. When you upgrade the .NET SDK this invariant is broken so the analyzer will start flagging any preview feature usage, be it language feature or API. The only time this won't result in a warning is when you had the flag set but weren't actually using a preview feature. 

This at least is how I understand this to work. Am I missing something here?

Reply:
No, you're correct and I was under a grave misapprehension: I didn't realize (or forgot) that the SDK actually enforces that `EnablePreviewFeatures` is only honored when the TFM matches the SDK's current framework. That certainly warrants a different language for this document.

Reply:
Addressed by [3831d17](https://github.com/dotnet/designs/pull/284/commits/3831d17ba2d312ba7ec522f634b3823d74864998)